be/src/vec/data_types/serde/data_type_serde.h

```diff
@@ -332,6 +336,11 @@ class DataTypeSerDe {
                                           Arena& mem_pool, int row_num) const;
     virtual Status read_one_cell_from_json(IColumn& column, const rapidjson::Value& result) const;
 
+    virtual DataTypeSerDeSPtrs get_nested_serdes() const {
```

Comment:
maybe we can support 
`get_key_serde()` `get_value_serde()` for map     
`get_data_serde()` for array  nullcolumn
`get_serde(int)` for struct 
not  `get_nested_serdes()` for any complex type